data/fin.lean

```diff
@@ -88,3 +91,23 @@ by cases i; refl
   {n} {C : fin (succ n) → Sort*} {H0 Hs} (i : fin n) :
   @fin.cases n C H0 Hs i.succ = Hs i :=
 by cases i; refl
+
+def nat_expand_type {n} (i : fin n) (k) : fin (n + k) :=
+⟨i.val,
+begin
+  have h : i.val < n, by apply i.is_lt,
+  apply lt_add_of_lt_of_nonneg,
+  apply h,
+  have h2 : k = 0 ∨ k > 0, by apply nat.eq_zero_or_pos,
+  rw or_comm at h2,
+  rw le_iff_lt_or_eq,
+  rw eq_comm,
+  apply h2,
+end
+⟩
+
+def contract_type_first {n m} (i : fin (n + m)) (h : i.val < n) : fin n :=
```

Comment:
Likewise, I would rename `contract_type_first` to `lower_left` and `contract_type_second` to `lower_right` using `lower` as the antonym of `raise`.

Reply:
Good idea.